pkg/network/common/egressip.go

```diff
@@ -249,6 +252,17 @@ func (eit *EgressIPTracker) handleAddOrUpdateHostSubnet(obj, _ interface{}, even
 			return
 		}
 	}
+	if eit.recorder != nil && len(hs.EgressIPs) > 0 && eit.CloudEgressIP {
+		capacity, err := eit.getCapacity(hs)
+		if err != nil {
+			klog.Errorf("Ignoring invalid HostSubnet %s: %v", HostSubnetToString(hs), err)
+		}
+		if len(hs.EgressIPs) >= capacity {
+			hsRef := &corev1.ObjectReference{Kind: "HostSubnet", Name: hs.Name, UID: hs.ObjectMeta.UID}
+			eit.recorder.Eventf(hsRef, corev1.EventTypeNormal, "HostSubnet",
+				"The Hostsubnet reached its maximum node capacity %d", capacity)
```

Comment:
So it's hard to figure out with the `pkg/network/common/egressip.go` code, but does this get logged by the master, the workers, or both? It should only get logged in one place; either by the master, or by the specific node that has reached capacity.

Also, we need to make sure that it doesn't spam events... we want an event when the node _reaches_ capacity, but we shouldn't log again after that unless it drops below capacity for a while and then hits it again.

Maybe a metric would be better than an Event, and then the admin can decide how to monitor the metric?

Reply:
@danwinship I guess this code always gets executed in node upon host subnet create and update and this node capacity event is triggered at the last egress ip. now added 5 mins interval gap to avoid spam. is this fine ? 
Moreover it's done here because event recorder is readily there from node code and consumed here  (master doesn't have event handler and it would just ignore this).